src/languages/swift.js

```diff
@@ -126,22 +127,98 @@ export default function(hljs) {
       SINGLE_LINE_STRING("###"),
     ]
   };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID412
+  const QUOTED_IDENTIFIER = {
+    begin: concat(/`/, Swift.identifier, /`/)
+  };
+  const IMPLICIT_PARAMETER = {
+    className: 'variable',
+    begin: /\$\d+/
+  };
+  const PROPERTY_WRAPPER_PROJECTION = {
+    className: 'variable',
+    begin: `\\$${Swift.identifierCharacter}+`
+  };
+  const IDENTIFIERS = [QUOTED_IDENTIFIER, IMPLICIT_PARAMETER, PROPERTY_WRAPPER_PROJECTION];
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Attributes.html
+  const AVAILABLE = {
+    begin: /(@|#)available/,
+    returnBegin: true,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /(@|#)available/
+      },
+      {
+        begin: /\(/,
+        end: /\)/,
+        endsParent: true,
+        keywords: Swift.availabilityKeywords.join(' '),
+        contains: [
+          OPERATOR,
+          NUMBER,
+          STRING,
+        ]
+      }
+    ]
+  };
+  const KEYWORD_ATTRIBUTE = {
+    className: 'keyword',
+    begin: concat(/@/, either(...Swift.keywordAttributes))
+  };
+  const USER_DEFINED_ATTRIBUTE = {
+    className: 'meta',
+    begin: concat(/@/, Swift.identifier)
+  }
+  const ATTRIBUTES = [
+    AVAILABLE,
+    KEYWORD_ATTRIBUTE,
+    USER_DEFINED_ATTRIBUTE
+  ];
+
+  const TYPE = {
+    className: 'type',
+    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*',
+    relevance: 0
+  };
+  // slightly more special to swift
+  const OPTIONAL_USING_TYPE = {
+    className: 'type',
+    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*[!?]'
```

Comment:
Doesn't TypeScript also have this same thing?

Reply:
Maybe, optional types are becoming more common now. 
But I didn't change this code, it just got moved around a bit.

Reply:
> But I didn't change this code, it just got moved around a bit.

No blame; you're pretty much rewritten the whole file (as you pointed out before) so now that means your also shining the light on some other broken things we should fix while we see them... that's just what happens in huge language reboot PRs.

So I don't think we need two rules here anymore... just a single rule with an optional `[?!]`, but `relevance: 0` for all cases.

Reply:
I've set the relevance to 0.